EDMarketConnector.py

```diff
@@ -1139,7 +1163,8 @@ def enumwindowsproc(window_handle, l_param):
                     and GetProcessHandleFromHwnd(window_handle):
                 # If GetProcessHandleFromHwnd succeeds then the app is already running as this user
                 if len(sys.argv) > 1 and sys.argv[1].startswith(protocolhandler.redirect):
-                    # Browser invoked us directly with auth response. Forward the response to the other app instance.
+                    logger.debug('Browser invoked us directly with auth response. '
```

Comment:
Will this write to our normal log file? I think it will. Possibly stepping in the middle of a write?

Reply:
The fun thing is in checking what happened with that ... now *this* code is:

```py
Exception ignored on calling ctypes callback function: <function enforce_single_instance.<locals>.enumwindowsproc at 0x04D2CBF8>
Traceback (most recent call last):
  File "C:/Users/Athan/Documents/Devel/EDMarketConnector/EDMarketConnector.py", line 1178, in enumwindowsproc
    sys.exit(0)
SystemExit: 0```
when it works fine with the Release/4.1.3 code.

Reply:
Now, is there any chance that Python 3.8.6 changed behaviour on this from 3.7.9 ?  i.e. is the solution to 'return False' and check that ?

Reply:
Well, that does seem to be working.

Reply:
And now I checked what that did with log files... yes, any logging from that will cause overwrite of the plain log file, but unless we adjust things that will have happened by this point now anyway, as we do the redirect, which truncates, ASAP.

The debug log file appears to work OK.  This is with 4.1.3 running and using this branch's code as second instance:

```py
2020-10-14 19:38:46.043 - DEBUG - companion.Session.start:455: Starting session
2020-10-14 19:41:38.786 - INFO - 17404:1004:1004 EDMarketConnector.__main__:1227: Startup v4.2.0-beta1 : Running on Python v3.8.6 (tags/v3.8.6:db45529, Sep 23 2020, 15:37:30) [MSC v.1927 32 bit (Intel)]
2020-10-14 19:47:43.320 - DEBUG - 16912:7884:7884 hotkey.HotkeyMgr.unregister:320: No thread
2020-10-14 19:47:43.528 - DEBUG - 16912:7884:7884 hotkey.HotkeyMgr.unregister:322: Done.
2020-10-14 19:49:46.415 - INFO - <plugins>.plugintest.load.plugin_stop:105: Stopping
2020-10-14 19:49:46.438 - INFO - EDMarketConnector.__main__:1204: Exiting
```
Note the two hotkey lines in the middle with process/thread IDs from this branch.  Just the import of hotkey is enough to cause that thread to start, and then main thread exit causes it to exit and log that.

So it *looks* like debug log file contents will end up done line by line, no overwrite.

Reply:
I can move the call to `enforce_single_instance()` to before even the redirect, and have no logging, *but* the point of this was to make it *obvious* when "installed new version, won't run" is because there's a lingering EDMarketConnector.exe process.

If we really don't want to overwrite EDMarketConnector.log in that case then I'd want to add a pop-up of some form in that case so the user is informed.  I'll take a look at the best way to do that tomorrow.

Reply:
Up to you if we overwrite, though I'd rather see about appending instead